p2panda-net/src/sync/mod.rs

```diff
@@ -53,12 +53,18 @@ where
     // Spawn a task which picks up any new application messages and sends them on to the engine
     // for handling.
     {
+        let engine_actor_tx = engine_actor_tx.clone();
         let mut sync_handshake_success = false;
         let topic = topic.clone();
 
         tokio::spawn(async move {
             while let Some(message) = rx.recv().await {
-                // We expect the first message to be a topic.
+                // 1. Handshake Phase.
+                // ~~~~~~~~~~~~~~~~~~~
+                //
+                // At the beginning of every sync session the "initiating" peer needs to send over
+                // the topic to the "accepting" peer during the handshake phase. This is the first
+                // message we're expecting:
                 if let FromSync::HandshakeSuccess(_) = &message {
                     if sync_handshake_success {
```

Comment:
Here we also end up with a protocol violation which gets silently caught

Reply:
Good catch. In this case we should send a `ToEngineActor::SyncFailed` event so the buffers can be unlocked and drained.